cli/src/pcluster/validators/tags_validators.py

```diff
@@ -14,18 +14,51 @@
 class ComputeResourceTagsValidator(Validator):
     """Compute resources tags validator."""
 
-    def _validate(self, queue_name, compute_resource_name, cluster_tags, queue_tags):
+    def _validate(self, queue_name, compute_resource_name, cluster_tags, queue_tags, compute_resource_tags):
         cluster_tags = {tag.key: tag.value for tag in cluster_tags} if cluster_tags else {}
         queue_tags = {tag.key: tag.value for tag in queue_tags} if queue_tags else {}
+        compute_resource_tags = {tag.key: tag.value for tag in compute_resource_tags} if compute_resource_tags else {}
+        overlapping_keys = set(cluster_tags.keys()) & set(queue_tags.keys()) & set(compute_resource_tags.keys())
+        overlapping_keys_list = sorted(list(overlapping_keys))
+        queue_cluster_overlapping_keys = sorted(
+            list(set(cluster_tags.keys()) & set(queue_tags.keys()) - overlapping_keys)
+        )
+        compute_resource_queue_overlapping_key = sorted(
+            list(set(queue_tags.keys()) & set(compute_resource_tags.keys()) - overlapping_keys)
+        )
+        cluster_compute_resource_overlapping_key = sorted(
+            list(set(cluster_tags.keys()) & set(compute_resource_tags.keys()) - overlapping_keys)
+        )
```

Comment:
The names are a little inconsistent here:
- overlapping_key**s_list**
- queue_cluster_overlapping_key**s**
- compute_resource_queue_overlapping_key
- cluster_compute_resource_overlapping_key

Maybe we can simplify this part a little bit? Something like:
```
cluster_keys = {tag.key for tag in cluster_tags}
queue_keys = {tag.key for tag in queue_tags}
compute_resource_keys = {tag.key for tag in compute_resource_tags}
overlapping_keys = cluster_keys & queue_keys & compute_resource_keys
...
```
We only need the tag keys not values here right?

Reply:
Yeah I was thinking mention the tag value in the validation errors, but we should be fine with just mentioning the tag keys, refactored the code here